### ISSUE TITLE
[keba] (2.5.11 and 3.0 RC1) did not work, items stay "NULL"

### ISSUE THREAD

**Issue by lgusewski** [openhab/openhab-addons#9420]
## Expected Behavior
After defining the thing

`Thing keba:kecontact:1 "KEBA KeContact P30" @ "Carport" [ipAddress="192.168.178.40", refreshInterval=30]`

it goes online and after defining items

`String      KebaState      {channel="keba:kecontact:1:state"}`

it should show an value.

## Current Behavior

The thing goes online (openHAB 2.5.11 and openHAB 3.0 RC1), but the items shows no value or NULL

## Possible Solution
network problems in the binding programming. Logfiles shows:

17:52:31.984 [DEBUG] [internal.handler.KeContactTransceiver] - Starting the the KEBA KeContact transceiver
17:52:31.990 [INFO ] [internal.handler.KeContactTransceiver] - Listening for incoming data on /0:0:0:0:0:0:0:0:7090
17:52:31.991 [DEBUG] [internal.handler.KeContactTransceiver] - Establishing the connection to the KEBA KeContact 'keba:kecontact:P30'
17:52:34.996 [DEBUG] [eba.internal.handler.KeContactHandler] - Ping timed out after '3001' milliseconds
17:52:34.999 [DEBUG] [internal.handler.KeContactTransceiver] - Stopping the the KEBA KeContact transceiver
17:52:36.903 [INFO ] [re.automation.internal.RuleEngineImpl] - Rule engine started.
17:52:53.004 [DEBUG] [eba.internal.handler.KeContactHandler] - Ping timed out after '3003' milliseconds
17:53:11.010 [DEBUG] [eba.internal.handler.KeContactHandler] - Ping timed out after '3003' milliseconds
17:53:29.013 [DEBUG] [eba.internal.handler.KeContactHandler] - Ping timed out after '3001' milliseconds

## Steps to Reproduce (for Bugs)

Install a new openHAB, install KEBA Binding and define the thing and an item. Should be give the problem. Firmware ob P30 is 1.11.0.

See also this [thread](https://community.openhab.org/t/keba-binding-not-working/110518) in the openHAB Community.

## Your Environment
All running with Ubuntu Server 20.04 LTS on two different VM.

openHAB 2.5.11 with Zulu Java 8
openHAB 3.0 RC with Zulu Java 11

**Comment by kaikreuzer** [comment]
Note that there is no general issue. I am using this binding since 1.5 years, always in its latest version and it always worked for me. Richt now I am on 3.0 RC1.
It might hence require a more detailed analysis of the network traffic on what exactly is wrong in your case.

**Comment by lgusewski** [comment]
> I am using this binding since 1.5 years

What type (series) of wallbox do you have? And which firmware? Perhaps there are the differences. [Here](https://community.openhab.org/t/openhab-3-0-release-candidates-discussion/110200/103) is somebody with the same problem.

My complete firmware (x-Series):

```
Release Version: 1.11.0

Component	Version
PDC	3.10.16
CPM	4.11.0
KEEP	5.4.0
OS	1.12.0
WEBUI	1.12.0
```

Because of the string "Listening for incoming data on /0:0:0:0:0:0:0:0:7090" I have some minutes before the idea that I could do something with IPv6, so I disabled IPv6 with these commands:

```
sudo sysctl -w net.ipv6.conf.all.disable_ipv6=1
sudo sysctl -w net.ipv6.conf.default.disable_ipv6=1
sudo sysctl -w net.ipv6.conf.lo.disable_ipv6=1
```

and restart openHAB with `sudo systemctl restart openhab`.

The result:

```
18:59:59.895 [DEBUG] [internal.handler.KeContactTransceiver] - Starting the the KEBA KeContact transceiver
18:59:59.897 [INFO ] [internal.handler.KeContactTransceiver] - Listening for incoming data on /0.0.0.0:7090
18:59:59.898 [DEBUG] [internal.handler.KeContactTransceiver] - Establishing the connection to the KEBA KeContact 'keba:kecontact:P30'
19:00:02.907 [DEBUG] [eba.internal.handler.KeContactHandler] - Ping timed out after '3002' milliseconds
19:00:02.909 [DEBUG] [internal.handler.KeContactTransceiver] - Stopping the the KEBA KeContact transceiver
19:00:06.429 [INFO ] [re.automation.internal.RuleEngineImpl] - Rule engine started.
19:00:20.914 [DEBUG] [eba.internal.handler.KeContactHandler] - Ping timed out after '3003' milliseconds
```

**Comment by lgusewski** [comment]
This example flow works in Node-RED 1.2.6 also on Ubuntu Server 20.04 LTS

`[{"id":"7616cee1.51669","type":"inject","z":"2469fb6e.6718f4","name":"","props":[{"p":"payload"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"report 2","payloadType":"str","x":100,"y":40,"wires":[["73041e67.abc38"]]}]`

Result:

`{"ID": "2","State": 1,"Error1": 0,"Error2": 0,"Plug": 3,"AuthON": 1,"Authreq": 1,"Enable sys": 0,"Enable user": 1,"Max curr": 0,"Max curr %": 1000,"Curr HW": 16000,"Curr user": 63000,"Curr FS": 0,"Tmo FS": 0,"Curr timer": 0,"Tmo CT": 0,"Setenergy": 0,"Output": 0,"Input": 0,"Serial": "********","Sec": 176604}`

Serial manually hidden.

**Comment by kaikreuzer** [comment]
I'll try to help on the investigation after christmas - won't manage to find any time before.
Feel free to ping me again in case I forget...

**Comment by lgusewski** [comment]
Made a test with Windows 10, Zulu Java 11 and openHAB 3.0 RC1. Here the Binding works. So it is really a Linux problem.

**Comment by kaikreuzer** [comment]
That's an interesting finding!
FTR: I am running openHAB on macOS.

**Comment by lgusewski** [comment]
Ok, more information, it is really crazy and i did not have ideas what I can do more. So last night/day i load Debian 10.7, Ubuntu Server 20.10 and macOS 10.14 and put them in virtual machines on my Mac in VMware Fusion. I make a minimal installation of Debian (no GUI), a minimal Ubuntu Server 20.10 (no GUI, only SSH Server added) and a normal macOS 10.14 installation with all updates at the end.

For Debian and Ubuntu, I choose the [Azul Zulu Java 11](https://docs.azul.com/zulu/zuludocs/ZuluUserGuide/InstallingZulu/InstallOnLinuxUsingDebRepository.htm) as JRE and I also tried JDK. For the macOS, I choose the same (filename zulu11.43.55-ca-jdk-11.09.1-macosx_x64.zip, put the folder zulu-11.jdk to /Library/Java/JavaVirtual/Machines/zulu-11.jdk and test in the terminal with java -version. Got result.

Then I load openHAB 3.0 RC1, start this in Debian 10.7 and Ubuntu 20.10 as as service (systemctl start openhab.service) and in macOS in the terminal with the ./start.sh script. Comment: One test after the other, not all machines at the same time!

Next step running the setup assistent steps, added the KEBA Binding, put in the IP-Address and added some items. In all three test (macOS included) I got again timeout in the log and no results.

Then I decided to make an additional test and load openHABian 1.6.1 for my Raspberry PI 3B+. After the installation I used here the default openHAB 2.5 I got in the test items values from the KEBA Wallbox. I looked also in the log files and saw that the binding seems to make the same UDP questions like I hat done with Node-RED.

I wonder that openHABian works, but Debian 10.7 not. I haven't changed anything in the network settings, only Ubuntu 20.10 got a static ip over the setup assistant, the others got there IP over DHCP.

I am frustrated

Ok, last information for this test: My productive system (Ubuntu Server 20.04.1 LTS) runs as a VM in ProxmoxVE 6.3 under KVM. Node-RED also, so I think when there is a problem with network, Node-RED should have this problem also.

**Comment by Jogobo** [comment]
As my Keba P30 C-Series shows the same behaviour (connection lost after a couple of hours) this information from the debug log might help.
After each report request ("1", "2", "3", "100") the report data is followed by this log message:

`^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@ from /192.xxx.xxx.xxx:7090 on the transceiver listener port
`

**Comment by MikeTheTux** [comment]
Is there a chane to get #6347 fixed?
I don't want to run OH as root.
There is no progress in the ticket since a year.
The solutions seems to be already present in 
https://github.com/Raketenschnitzel/openhab2-addons/commit/10973ca38b90dcbdbf9861a9008c1de4f4b95e0e#diff-4fc5c91a270b28eb3ea2236c2e2c6be1a6dfff9ac7fb649412d243dae2910031

**Comment by Jogobo** [comment]
As a workaround I restart the binding every 6 hours with 
`sshpass -p <my_openhab_cli_password> ssh -p 8101 openhab@localhost "bundle:restart <id_of_keba_binding>"`

**Comment by MikeTheTux** [comment]
Please have a look at [#6347](https://github.com/openhab/openhab-addons/issues/6347#issuecomment-778670248) where I posted an OH3 compatible hotfix of the Keba binding

**Comment by Jogobo** [comment]
Will not help yet as I am still on 2.5 and will be for at least the next 6 months.

**Comment by codefalk** [comment]
Same problem here. Thanks @MikeTheTux for the fix. 
However, I am also still on 2.5.12. Is there a chance to have the issue fixed for 2.5 as well?

**Comment by kaikreuzer** [comment]
@codefalk Unfortunately not. There are no further 2.5.x releases planned unless necessary for critical security fixes.

**Comment by lsiepel** [comment]
I assume this issue can be closed as it is fixed since openHAB 3.1.0 and the 2.5 branch is no longer supported. If there is still an issue with 3.4.0, please feel free to re-open.
